clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -350,6 +350,11 @@ class PointerVariableConstraint : public ConstraintVariable {
 
   ConstraintVariable *getCopy(Constraints &CS);
 
+  // Retrieve the atom at the specified index. This function includes special
+  // handling for generic constraint variables to create deeper pointers as
+  // they are needed.
+  Atom *getAtom(unsigned int AtomIdx, Constraints &CS);
```

Comment:
What is an "index?" Is this the index into the `cvars` array? As we add to the "API" for accessing this thing, I wonder if we can get rid of `getCvars`, to encapsulate it better? (Could do that as a future code improvement.)

Reply:
Yes, this function is used to access `vars`. It would be nice to avoid directly accessing it through `getCVars`, but that does at least return a const reference, providing some protection against misuse.